source/lib.civet

```diff
@@ -6,7 +6,8 @@
  * here eventually.
  */
 
-type ASTNode = ASTNodeBase | ASTError | ASTRef | ASTLeaf | ASTNode[] | string | undefined
+type ASTNode = ASTNodeBase | ASTError | ASTRef | ASTLeaf | ASTNode[] | ASTString | undefined
+type ASTString = string & ASTNodeBase
```

Comment:
I don't understand this type definition... Strings don't have type etc. right? I could see adding `type: never, children: never` though.

Reply:
It's a hack to fix the type errors on gather .type is "SomeNode" but it causes errors when inserting strings so probably a wash.

Reply:
Ah, I see. Maybe `type?: never` would be a better hack for strings? I think this would enable checking but without actually allowing it to exist...